flare/mgp/mapxb.py

```diff
@@ -0,0 +1,655 @@
+            mean_only = True
+
+        force_kernel, en_kernel, _, cutoffs, hyps, hyps_mask = self.kernel_info
+
+        args = from_mask_to_args(hyps, cutoffs, hyps_mask)
+
+        kern = 0
+        if not mean_only:
+            if self.map_force:
+                kern = np.zeros(3)
+                for d in range(3):
+                    kern[d] = force_kernel(atom_env, atom_env, d+1, d+1, *args)
+            else:
+                kern = en_kernel(atom_env, atom_env, *args)
+
+        spcs, comp_r, comp_xyz = self.get_arrays(atom_env)
+
+        # predict for each species
+        f_spcs = np.zeros(3)
+        vir_spcs = np.zeros(6)
+        v_spcs = np.zeros(3) if self.map_force else 0
+        e_spcs = 0
+        for i, spc in enumerate(spcs):
+            lengths = np.array(comp_r[i])
+            xyzs = np.array(comp_xyz[i])
+            map_ind = self.find_map_index(spc)
+
+            f, vir, v, e = self.maps[map_ind].predict(lengths, xyzs,
+                self.map_force, mean_only)
+            f_spcs += f
+            vir_spcs += vir
+            v_spcs += v
+            e_spcs += e
+
+        return f_spcs, vir_spcs, kern, v_spcs, e_spcs
+
+    def as_dict(self) -> dict:
+        """
+        Dictionary representation of the MGP model.
+        """
+
+        out_dict = deepcopy(dict(vars(self)))
+
+        # Uncertainty mappings currently not serializable;
+        if not self.mean_only:
+            out_dict['mean_only'] = True
+
+        # save the kernel name instead of callables
+        kernel, ek, efk, cutoffs, hyps, hyps_mask = self.kernel_info
+        out_dict['kernel_info'] = (kernel.__name__, ek.__name__, efk.__name__,
+                               cutoffs, hyps, hyps_mask)
+
+        # only save the mean coefficients
+        out_dict['maps'] = [m.mean.__coeffs__ for m in self.maps]
+        out_dict['bounds'] = [m.bounds for m in self.maps]
+
+        # rm keys since they are built in the __init__ function
+        key_list = ['singlexbody', 'spc_set']
+        for key in key_list:
+            if out_dict.get(key) is not None:
+                del out_dict[key]        
+
+        return out_dict
+
+    @staticmethod
+    def from_dict(dictionary: dict, mapxbody):
+        """
+        Create MGP object from dictionary representation.
+        """
+
+        # Set GP
+        if dictionary.get('GP'):
+            GP = GaussianProcess.from_dict(dictionary.get("GP"))
+        else:
+            dictionary['GP'] = None
+
+        if 'container_only' not in dictionary:
+            dictionary['container_only'] = True
+
+        # initialize
+        init_args_name = ['grid_num', 'lower_bound', 'upper_bound', 'svd_rank',
+            'species_list', 'map_force', 'GP', 'mean_only', 'container_only', 
+            'lmp_file_name', 'load_grid', 'lower_bound_relax', 'n_cpus', 'n_sample']
+        args = [dictionary[name] for name in init_args_name]
+        new_mgp = mapxbody(args)
```

Comment:
I think the above block could be replace by 
```python
new_mgp = mapxbody(**args)
```
So you don't need to maintain two argument lists (here and the init functions)